src/core/mesh/qgstriangularmesh.cpp

```diff
@@ -169,25 +169,31 @@ bool QgsTriangularMesh::update( QgsMesh *nativeMesh, const QgsCoordinateTransfor
 {
   Q_ASSERT( nativeMesh );
 
+  bool needUpdateVerticesCoordinates = mTriangularMesh.vertices.size() != nativeMesh->vertices.size() ||
+                                       ( ( mCoordinateTransform.isValid() || transform.isValid() ) &&
+                                         ( mCoordinateTransform.sourceCrs() != transform.sourceCrs() &&
```

Comment:
This assumption doesn't hold unfortunately -- you'd actually need to check the proj pipeline for the two transforms to check equality

Reply:
@nyalldawson, thanks to pointing that. Do you know if there is something in QGIS that uses this check with the good way?

Reply:
I'll have to dig... we probably should add a proper equality operator for QgsCoordinateTransform to handle this...

Reply:
I looked quickly how to have a proper equality operator, and it not seems easy, maybe because I don't now a lot about this part of QGIS...
I understand it is not a good solution, but so far, that worked this way... maybe we could let like this waiting for a proper equality operator?

Reply:
I think to have proper equality operator would be great....